app/src/main/java/ch/epfl/sweng/SDP/auth/Account.java

```diff
@@ -262,33 +264,70 @@ private void updateCurrentLeague() {
-                    @Override
-                    public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
-                        Long value = dataSnapshot.getValue(Long.class);
-                        if (value != null) {
-                            final int newStars = amount + value.intValue();
 
-                            DatabaseReferenceBuilder starsBuilder = new DatabaseReferenceBuilder(
-                                    usersRef);
-                            starsBuilder.addChildren(userId + ".stars").build()
-                                    .setValue(newStars, createCompletionListener());
-                            stars = newStars;
+        DatabaseReferenceBuilder starsBuilder = new DatabaseReferenceBuilder(
+                usersRef);
+        starsBuilder.addChildren(userId + ".stars").build()
+                .setValue(newStars, createCompletionListener());
+        stars = newStars;
 
-                            localDbHandler.saveAccount(instance);
-                        }
-                    }
+        localDbHandler.saveAccount(instance);
+    }
 
-                    @Override
-                    public void onCancelled(@NonNull DatabaseError databaseError) {
-                        throw databaseError.toException();
-                    }
-                });
+    /**
+     * Method that allows one to increase the number of matches won.
+     *
+     * @throws DatabaseException in case write to database fails
+     */
+    public void increaseMatchesWon()
+            throws IllegalArgumentException, DatabaseException {
+        DatabaseReferenceBuilder builder = new DatabaseReferenceBuilder(
+                usersRef);
+        builder.addChildren(userId + ".matchesWon").build()
+                .setValue(++matchesWon, createCompletionListener());
+
+        localDbHandler.saveAccount(instance);
+    }
+
+    /**
+     * Method that allows one to increase the number of matches lost.
+     *
+     * @throws DatabaseException in case write to database fails
+     */
+    public void increaseMatchesLost()
+            throws IllegalArgumentException, DatabaseException {
+        DatabaseReferenceBuilder builder = new DatabaseReferenceBuilder(
+                usersRef);
+        builder.addChildren(userId + ".matchesLost").build()
+                .setValue(++matchesLost, createCompletionListener());
+
+        localDbHandler.saveAccount(instance);
+    }
+
+    /**
+     * Method that allows one to change the average rating given a new rating.
+     *
+     * @throws DatabaseException in case write to database fails
+     */
+    public void changeAverageRating(double rating)
+            throws IllegalArgumentException, DatabaseException {
+        if (!(0 < rating && rating <= 5)) {
+            throw new IllegalArgumentException("Wrong rating given");
+        }
+
+        double newAverageRating = averageRating == 0 ? rating : (averageRating + rating) / 2.0;
```

Comment:
The new average is incorrect, change this by (averageRating * (totalMatches - 1) + rating) / totalMatches
(Also, replace the statistic matchesLost by totalMatches)